src/main/java/core/basesyntax/db/FruitShopDaoImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.db;
+
+import core.basesyntax.storage.Storage;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitShopDaoImpl implements FruitShopDao {
+    @Override
+    public int storageSize() {
+        return Storage.fruitStorage.size();
+    }
+
+    @Override
+    public List<String> getKeyAndValue() {
+        List<String> lines = new ArrayList<>();
+        for (String key : Storage.fruitStorage.keySet()) {
+            Integer value = Storage.fruitStorage.get(key);
+            lines.add(key + ", " + value);
+        }
+        return lines;
+    }
+
+    @Override
+    public void put(String fruit, int quantity) {
+        Storage.fruitStorage.put(fruit, quantity);
+    }
+
+    @Override
+    public void putOnlyFruits(String fruit) {
+        Storage.fruitStorage.put(fruit, null);
```

Comment:
Using `null` as a value in the `Storage.fruitStorage` map is not a good practice. It can lead to `NullPointerExceptions` when other parts of the code try to read and manipulate these values. You should consider using an Optional or a special value to represent the absence of a quantity.